internal/clients/http/image_builder/image_client.go

```diff
@@ -50,9 +50,12 @@ func (c *ibClient) Ready(ctx context.Context) error {
 		logger.Error().Err(err).Msg("Readiness request failed for image builder")
 		return err
 	}
+	if resp == nil {
+		return fmt.Errorf("ready call: response not available")
+	}
```

Comment:
I don't have a problem with that, but I am wondering is that possible when err is nil?

Reply:
Here is docs of the client Do method:

https://pkg.go.dev/net/http#Client.Do

Here is the generated code:

https://github.com/RHEnVision/provisioning-backend/blob/main/internal/clients/http/image_builder/client.gen.go#L1088

My quick conclusion is that response cannot be nil, unless there was an error.

Reply:
Yeah, sure. That makes sense. I was just thinking...of course, in the Client.Do documentation we know this is how the function works. And we should probably be able to depend on that. I was thinking if this was not redundant and we could have a conversation on that for sure!:)
In this case, however, we are using a different function from a generated code,. I did not find any docs that would guarantee us that those functions, if the template possible changes, would have the same behaviour that we expect from the Client.Do.
Any thoughts on this or links to such docs?

Reply:
Good thinking, I am not against checking it.